codefresh-io/terraform-provider-codefresh
# Update Spec with Steps and Stages when OriginalYamlString is defined

Edit: I've pushed a second commit to address an issue with the attribute `steps` where the order of the steps was lost during the marshalling of the object. This happens because maps in JSON are unordered. I've updated the code to dynamically build the JSON string by parsing the original_yaml_string attribute to avoid losing the correct order

# Issue Description
When defining a pipeline using the attribute `original_yaml_string`, running the pipeline would result in an empty execution (no steps and stages)
Steps and stages are visible from the console editor after the pipeline is created, however, the `Spec.Stages` and `Spec.Steps` are not updated causing the pipeline to be empty during execution.

Documentation references:
[Original_yaml_string](https://codefresh.io/docs/docs/integrations/codefresh-api/#metadata-fields)
```
originalYamlString | metadata | string | the full contents of the pipeline editor. Only kept for archival purposes
```
[Steps and Stages](https://codefresh.io/docs/docs/integrations/codefresh-api/#spec-fields)
```
steps | spec | object | The pipeline steps to be executed
stages | spec | array | The pipeline stages for a better visual overview

``` 

# Steps to reproduce
1. Create a pipeline using the provider and leverage the `original_yaml_string` attribute
```
----pipeline.yaml ---
version: "1.0"
stages:
  - checkout
  - hello_world
steps:
  main_clone:
    title: Cloning main repository...
    type: git-clone
    repo: ${{CF_REPO_OWNER}}/${{CF_REPO_NAME}}
    revision: ${{CF_REVISION}}
    stage: checkout
  hello_world:
    stage: hello_world
    title: hello_world
    image: alpine:latest
    commands:
       - echo "Hello World"
---------

resource "codefresh_project" "test" {
  name = "myproject"
}

resource "codefresh_pipeline" "test" {
  name    = "${codefresh_project.test.name}/original_yaml_string_pipeline"

  tags = [
    "test",
  ]
 original_yaml_string = file("pipeline.yaml")
 spec {
    concurrency = 1
    priority    = 5
 }
```
2. Check the Pipeline in the UI, the editor displays all the steps
3. Execute the pipeline, no steps/stages are displayed nor executed
